### PR TITLE
Add TLS to the dummy typha client.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Found this useful when doing some scale testing and prototyping recently.

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: fv-tests/server_test.go [hunk 1]
```diff
@@ -287,6 +287,11 @@ var _ = Describe("With an in-process Server", func() {
 			expectedEndState[path] = update
 			decoupler.OnUpdates([]api.Update{update})
 		}
+		return expectedEndState
+	}
+
+	sendNUpdatesThenInSync := func(n int) map[string]api.Update {
+		expectedEndState := sendNUpdates(n)
```
Comment (neiljerram) [inline fv-tests/server_test.go:294]: Why this change?

Comment (fasaxc) [reply fv-tests/server_test.go:294]: I made a test locally that used this change but I didn't think it was worth pushing in its current form.  OTOH, this small refactoring would allow that kind of test to be written more easily in future so I kept it.
